From 95ecf362fb70cd5827d99c63760a4dad9b9af636 Mon Sep 17 00:00:00 2001
From: Alan Zhao <ayzhao@google.com>
Date: Thu, 22 Sep 2022 23:46:21 -0700
Subject: [PATCH] Disable several clang warnings

These silences are required to get 7z to build with Clang. The
alternative, which is to fix the code, is unwieldy because these errors
trigger in so many places that the resulting patch files would become
very unmanagable.  Additionally, the 7z source code doesn't have a Git
repository, and there are no known ways to contribute fixes upstream.
---
 CPP/7zip/warn_clang.mak | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CPP/7zip/warn_clang.mak b/CPP/7zip/warn_clang.mak
index a299fbc..75a74b3 100755
--- a/CPP/7zip/warn_clang.mak
+++ b/CPP/7zip/warn_clang.mak
@@ -30,6 +30,12 @@ CFLAGS_WARN_CLANG_12= $(CFLAGS_WARN_CLANG_3_8) \
 
 CFLAGS_WARN_1 = \
   -Wno-deprecated-copy-dtor \
+  -Wno-unknown-warning-option \
+  -Wno-reserved-identifier \
+  -Wno-suggest-override \
+  -Wno-suggest-destructor-override \
+  -Wno-declaration-after-statement \
+  -Wno-unused-but-set-variable \
 
 
 
-- 
2.37.3.998.g577e59143f-goog

